BookAPI/Startup.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace BookAPI
+{
+    using Microsoft.AspNetCore.Builder;
+    using Microsoft.AspNetCore.Hosting;
+    using Microsoft.Extensions.DependencyInjection;
+    using BookAPI.Services;
+    using Microsoft.AspNetCore.Mvc;
+
+    public class Startup
+    {
+        // This method gets called by the runtime. Use this method to add services to the container.
+        // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddSingleton<IBookShelf, BookShelf>();
```

Comment:
Try to not using singleton in the future.

Reply:
I have to use it because of Id incrementing. If i use another type of service like scoped or transient, id will be incrementing every time i address to list of books.